### PR TITLE
kyma_controller: Add Kyma CR Conditions Whitelist

### LINKED ISSUE #509: Refactor status.InitConditions function to handle with any future condition type will be deprecated.
At the moment, the `status.InitConditions` can only handle `DeprecatedConditionTypeReady` which is hardcoded, and can only handle with this specific condition. The better way is to write a function that is more generic, that we can handle with any condition in the future will be deprecated.
Idea: whitelist our current support condition type, and condition not in the whitelist should be removed during init condition.

AC:
- [ ] the InitConditions can remove any condition which consider being deprecated in the future.

### PR DESCRIPTION
**Pull request by LeelaChacha**
Closes #509

### CODE REVIEW

File: pkg/status/conditions.go [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
-		apimeta.RemoveStatusCondition(existingCondition, string(v1beta1.DeprecatedConditionTypeReady))
-	}
+	removeDeprecatedConditions(kyma)
 	// Add required Conditions
 	for _, cond := range v1beta1.GetRequiredConditionTypes(kyma.Spec.Sync.Enabled, watcherEnabled) {
 		kyma.UpdateCondition(cond, metav1.ConditionFalse)
 	}
 }
+
+func removeDeprecatedConditions(kyma *v1beta1.Kyma) {
```
[the remark this answers is not in the document]
Comment (LeelaChacha) [reply pkg/status/conditions.go:19]: Unit Test Implemented for InitConditions function. Test is passing

File: controllers/kyma_controller_test.go [hunk 2]
```diff
@@ -43,13 +41,22 @@ var _ = Describe("Kyma with deprecated Condition", Ordered, func() {
...
 			LastTransitionTime: metav1.Time{Time: time.Now()},
 		})
-		kyma.ManagedFields = nil
-		Eventually(controlPlaneClient.Patch(ctx, kyma, client.Apply,
-			client.FieldOwner(v1beta1.UnmanagedKyma)), Timeout*2, Interval).Should(Succeed())
-		By("having transitioned the CR State to Ready as there are no modules")
-		Eventually(CheckKymaConditions(ctx, controlPlaneClient, kyma.GetName(),
-			[]v1beta1.KymaConditionType{v1beta1.ConditionTypeModules}),
-			Timeout, Interval).Should(BeTrue())
+		kyma.Status.Conditions = append(kyma.Status.Conditions, metav1.Condition{
```
Comment (ruanxin) [inline controllers/kyma_controller_test.go:44]: This test is fine, but in this test, you only test the `InitConditions`, which can be written with a simple unit test, and doesn't need to be based on Ginkgo framework.

You can reference this [unit test example](https://github.com/kyma-project/lifecycle-manager/blob/main/internal/manifest/v1beta1/spec_resolver_test.go) to check how to write a unit test, and please also reference this [goland test guide](https://www.jetbrains.com/help/go/testing.html), it offers us a feature to automatically generate good unit test template to unit test the functions.

Comment (LeelaChacha) [reply controllers/kyma_controller_test.go:44]: But Ginkgo is a general framework for testing in Golang. I think it is better to stick to one framework for consistency

**Review by ruanxin — APPROVED** [review]

**Pull request merged**